### PR TITLE
Add functional interface for RMSNorm

### PR DESCRIPTION
**Pull request by t-vi**
emulating how PyTorch splits this for leaf modules.

### CODE REVIEW

**Review by rasbt — APPROVED** [review]

**Comment by Andrei-Aksionov** [comment]
Sorry for the dummy question, but why is it needed?

**Comment by t-vi** [comment]
It makes reuse and such easier because it disentangles the parameter storing from the compute. If you look at PyTorch, it also follows this.
I'll have a concrete example later this week.

**Comment by lantiga** [comment]
It makes it simpler for litgpt to be optimized with custom kernels, as you can replace the functional call more easily

**Comment by Andrei-Aksionov** [comment]
Thanks @lantiga and @t-vi for the explanation.

So that's to make it easier for LitGPT to go brr.
Funny, last week I started to read PMPP book for that exact goal 😄.

> I'll have a concrete example later this week.

Looking forward to it.

File: litgpt/model.py [hunk 1]
```diff
@@ -392,32 +392,45 @@ def build_mask_cache(max_seq_length: int, device: Optional[torch.device] = None)
     return torch.tril(ones).unsqueeze(0).unsqueeze(0)
 
 
-class RMSNorm(torch.nn.Module):
+def rms_norm(x: torch.Tensor, weight, *, dim: int, eps: float, add_unit_offset: bool) -> torch.Tensor:
```
Comment (Andrei-Aksionov) [inline litgpt/model.py:395]: Missing type hint for the `weight` argument.

Comment (rasbt) [inline litgpt/model.py:395]:
```suggestion
def rms_norm(x: torch.Tensor, weight: torch.Tensor, *, dim: int, eps: float, add_unit_offset: bool) -> torch.Tensor:
```

Good catch

**Comment by t-vi** [comment]
Ah, too late already. Thank you @rasbt @lantiga @Andrei-Aksionov

**Pull request closed without merging**